File: model/ServiceProvider.java
package com.servitrust.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "service_providers")
public class ServiceProvider {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "auth_id", nullable = false, unique = true)
    private ProviderAuth auth;

    @NotBlank
    private String name;

    @NotBlank
    private String serviceType;

    @NotBlank
    private String location;

    private String availabilityStatus = "AVAILABLE";
    private Double rating = 0.0;
    private Integer completedJobs = 0;
    private Integer acceptedCount = 0;
    private Integer cancelledCount = 0;
    private Double trustScore = 50.0;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "verified")
    private Boolean verified = false;

    public Boolean getVerified() { return verified; }
    public void setVerified(Boolean verified) { this.verified = verified; }

    public ServiceProvider() {}

    public Long getId() { return id; }

    public ProviderAuth getAuth() { return auth; }
    public void setAuth(ProviderAuth auth) { this.auth = auth; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getServiceType() { return serviceType; }
    public void setServiceType(String serviceType) { this.serviceType = serviceType; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }

    public String getAvailabilityStatus() { return availabilityStatus; }
    public void setAvailabilityStatus(String availabilityStatus) { this.availabilityStatus = availabilityStatus; }

    public Double getRating() { return rating; }
    public void setRating(Double rating) { this.rating = rating; }

    public Integer getCompletedJobs() { return completedJobs; }
    public void setCompletedJobs(Integer completedJobs) { this.completedJobs = completedJobs; }

    public Integer getAcceptedCount() { return acceptedCount; }
    public void setAcceptedCount(Integer acceptedCount) { this.acceptedCount = acceptedCount; }

    public Integer getCancelledCount() { return cancelledCount; }
    public void setCancelledCount(Integer cancelledCount) { this.cancelledCount = cancelledCount; }

    public Double getTrustScore() { return trustScore; }
    public void setTrustScore(Double trustScore) { this.trustScore = trustScore; }

    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }

    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }
}

File: model/ServiceProvider.kt
package com.servitrust.app.model

import com.google.gson.annotations.SerializedName

data class ServiceProvider(
    val id: Long? = null,
    val name: String,

    @SerializedName(value = "serviceType", alternate = ["service_type"])
    val serviceType: String,

    val location: String,

    @SerializedName(value = "availabilityStatus", alternate = ["availability_status"])
    val availabilityStatus: String? = null,

    val rating: Double? = 0.0,

    @SerializedName(value = "completedJobs", alternate = ["completed_jobs"])
    val completedJobs: Int? = 0,

    @SerializedName(value = "acceptedCount", alternate = ["accepted_count"])
    val acceptedCount: Int? = 0,

    @SerializedName(value = "cancelledCount", alternate = ["cancelled_count"])
    val cancelledCount: Int? = 0,

    @SerializedName(value = "trustScore", alternate = ["trust_score"])
    val trustScore: Double? = 0.0,

    @SerializedName(value = "latitude", alternate = ["lat"])
    val latitude: Double? = null,

    @SerializedName(value = "longitude", alternate = ["lng"])
    val longitude: Double? = null,

    val verified: Boolean? = false
)